File: src/main/resources/gemini-app/src/app/DTOs/QuestaoDTO.ts
import { Resposta } from '../model/resposta';

export class QuestaoDTO {
  constructor(
    public id: Number,
    public idquest: Number,
    public name: string,
    public alternativas: string,
    public resposta: string,
    public tipo_alternativa: Number,
    public listaRespostasQuestao: Resposta[]
  ) {}
}

File: src/main/java/com/example/Gemini/DTO/QuestaoDTO.java
package com.example.Gemini.DTO;

import java.util.List;

import com.example.Gemini.model.Resposta;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class QuestaoDTO {

    private long id;

    private long idquest;

    private String name;

    private String alternativas; //alternativas juntas aqui e pra separar usar função split("")

    private String resposta;

    private int tipo_alternativa;
    List<Resposta> listaRespostasQuestao;
}